XRDXRFutils/gaussnewton.py

```diff
@@ -364,17 +357,60 @@ def overlap3(self):
     def overlap3_area(self):
         return self.overlap3().sum()
 
-    def make_phase(self):
 
-        mu, I = self.get_theta()
-        new_I = I * self.gamma[0]
-        new_I /= new_I.max()
-        
-        new_phase = PhaseList(self.phase)
-        
-        new_phase.theta = mu
-        new_phase.intensity = new_I
-        
-        #new_phase.label = self.label
+    def make_phases(self):
+        """
+        Creates experimental phases from the phases used to create the given instance of GaussNewton.
+        An instance of GaussNewton can be created with a Phase or a PhaseList passed as the argument 'phase'.
+        In any case, this function returns a PhaseList of the experimental phases.
+
+        Please, note that this function exludes peaks that lay outside the angle range of experimental signal.
+        This is because they can have anomalously high value in order to fit the last bit of signal. This dwarfs all the other peaks.
+        Also, if the given instance of GaussNewton is created with a value of 'min_intensity', this function selects the peaks with relative intensity above that threshold.
+        If a phase has no selected peaks, it is not added to the results.
+        """
+        if type(self.phase) == Phase:
+            pl = PhaseList([self.phase])
+            is_pl = False
+        elif type(self.phase) == PhaseList:
+            pl = self.phase
+            is_pl = True
+        else:
+            raise Exception('Invalid phase type')
+
+        pl_new = PhaseList([])
+        index_count = 0
+
+        for phase in pl:
+            mu, I = phase.get_theta()
+            phase_len = mu.shape[0]
+
+            gamma = self.gamma.squeeze()[index_count : (index_count + phase_len)]
+            index_count += phase_len
+
+            I_new = I * gamma
+            I_new /= I_new.max()
+
+            # Selects only the peaks that lay inside the angle range of experimental signal.
+            # Otherwise, external peaks can have anomalously high value in order to fit the last bit of signal. This dwarfs all the other peaks.
+            theta_min, theta_max = self.theta_range()
+            mask_theta = ((mu >= theta_min) & (mu <= theta_max))
+            mu = mu[mask_theta]
+            I_new = I_new[mask_theta]
+
+            # Selects only the peaks with relative intensity above the chosen threshold.
+            if 'min_intensity' in self.kwargs:
```

Comment:
This should not be there. You should filter this when you load the phase not when you create it or write it to file.